### ISSUE TITLE
Killing Sync/Build on VS Code Exit

### ISSUE THREAD

**Issue by dsanders11** [dsanders11/vscode-electron-build-tools#4]
Currently a sync or build will continue if you exit the window, which is not ideal. Figure out the best way to kill them off when VS Code exits.

**Comment by dsanders11** [comment]
Running it as a task lets VS Code handle all the icky killing business, which is nice, and it can put the output into a terminal where the user can still see the raw output. However, there's no way to access stdout, which is needed to be able to implement progress indication. It may be possible to pipe the output to a named pipe or socket and then listen to that. Then to maintain the terminal stuff it'd be piped back into the terminal, which is easy enough. See https://nodejs.org/api/net.html#net_ipc_support.

```typescript
vscode.commands.registerCommand("electron-build-tools.build", () => {
  const command = "electron-build-tools build";
  const operationName = "Electron Build Tools - Building";

  const writeEmitter = new vscode.EventEmitter<string>();
  const pty: vscode.Pseudoterminal = {
    onDidWrite: writeEmitter.event,
    open: () => {},
    close: () => {},
    handleInput: data => writeEmitter.fire(data === '\r' ? '\r\n' : data)
  };

  const task = new vscode.Task(
    { type: 'electron-build-tools', task: 'build' },
    vscode.workspace.workspaceFolders![0],
    'build',
    'electron-build-tools',
    new vscode.CustomExecution(async () => {
      return pty;
    })
  );

  task.presentationOptions = {
    'reveal': vscode.TaskRevealKind.Silent,
    'panel': vscode.TaskPanelKind.Dedicated
  };

  vscode.window.withProgress(
    {
      location: vscode.ProgressLocation.Notification,
      title: operationName.split("-")[1].trim(),
      cancellable: true,
    },
    async (progress, token) => {
      const executingTask = await vscode.tasks.executeTask(task);
      token.onCancellationRequested(() => {
        executingTask.terminate();
      });

      return new Promise(resolve => {
        vscode.tasks.onDidEndTask((foo) => {
          if (foo.execution === executingTask) {
            resolve();
          }
        });
      });
    })
  ;
}),
```

**Comment by dsanders11** [comment]
Fixed in e83751ef7301af65b27d88b3d28669afc11ac79b.
